### PR TITLE
Transform map[something]SomethingNotNullable

### PR DESCRIPTION
**Pull request by Serabe**
Add the option to transform `map[x]Y` correctly.

It also adds some testing to `protobuf.Type` `IsNullable` method.

Resolving aliases had some problems too.

### CODE REVIEW

**Comment by Serabe** [comment]
Except `generated.pb.go`, everything else is reviewable.

File: protobuf/protobuf_test.go [hunk 1]
```diff
@@ -52,3 +54,57 @@ func TestOptionsString(t *testing.T) {
 	require.Equal(t, "foo", NewLiteralValue("foo").String())
 	require.Equal(t, `"bar"`, NewStringValue("bar").String())
 }
+
+func TestIsNullable(t *testing.T) {
+	var typ Type
```
Comment (dpordomingo) [inline protobuf/protobuf_test.go]: why not using a foreach to test this bunch of nullables?

Comment (Serabe) [inline protobuf/protobuf_test.go]: Because the logic is not the same, some depend uniquely on the source, some does not, etc.

**Comment by Serabe** [comment]
Rebased, waiting for travis.

File: protobuf/protobuf_test.go (hunk 1 above)
Comment (dpordomingo) [inline protobuf/protobuf_test.go]: ok, fair enough

**Pull request merged**